app/services/event_disavowal/find_disavowed_event.rb

```diff
@@ -15,11 +15,7 @@ def call
     private
 
     def event
-      # Use `#all` here instead of `#first` to avoid setting a 'LIMIT 1' to the
-      # postgres query which causes it to run slowly.
-      @event ||= Event.where(
-        disavowal_token_fingerprint: disavowal_token_fingerprints,
-      ).all[0]
+      @event ||= Event.find_by(disavowal_token_fingerprint: disavowal_token_fingerprints)
```

Comment:
as long as we're here optimizing, should we do a `defined?` check so that we don't risk running the query multiple times with a `nil` result?
```suggestion
      return @event if defined?(@event)
      @event = Event.find_by(disavowal_token_fingerprint: disavowal_token_fingerprints)
```

Reply:
Yea that's a good idea 👍